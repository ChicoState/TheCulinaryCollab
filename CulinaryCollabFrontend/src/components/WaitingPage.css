
h2 {
    font-size: 3em;
    margin-bottom: 20px;
}
