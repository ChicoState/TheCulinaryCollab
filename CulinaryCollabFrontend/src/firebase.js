// Import the functions you need from the SDKs you need
import { initializeApp } from "firebase/app";
import { getAnalytics } from "firebase/analytics";
import { getFirestore } from "firebase/firestore";
// TODO: Add SDKs for Firebase products that you want to use
// https://firebase.google.com/docs/web/setup#available-libraries

// Your web app's Firebase configuration
// For Firebase JS SDK v7.20.0 and later, measurementId is optional
const firebaseConfig = {
  apiKey: "AIzaSyDE3ZbA90p3yVdRCjiU_LgmUJOXXe_1dso",
  authDomain: "culinarycollab.firebaseapp.com",
  projectId: "culinarycollab",
  storageBucket: "culinarycollab.appspot.com",
  messagingSenderId: "908985843966",
  appId: "1:908985843966:web:fecba0089ca2c22bd4e5af",
  measurementId: "G-700XG3Z51G"
};

// Initialize Firebase
const app = initializeApp(firebaseConfig);
const analytics = getAnalytics(app);
export const firestore = getFirestore(app);
