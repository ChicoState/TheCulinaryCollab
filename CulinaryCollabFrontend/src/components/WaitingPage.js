import React, { useEffect } from 'react';
import { auth } from '../firebase';
import { useNavigate } from 'react-router-dom';

const WaitingPage = () => {
  const navigate = useNavigate();

  useEffect(() => {
    const checkEmailVerification = async () => {
      try {
        await auth.onAuthStateChanged(async (user) => {
          if (user) {
            await user.reload();
            if (user.emailVerified) {
              // Email verified, navigate to the desired page
              navigate('/workshop');
            }
          }
        });
      } catch (error) {
        console.error('Error checking email verification: ', error);
      }
    };

    const interval = setInterval(checkEmailVerification, 3000); // Check every 3 seconds
    return () => clearInterval(interval);
  }, [navigate]);

  return (
    <div>
      <h2>Please wait while we verify your email...</h2>
      {/* You can add a loader or any other waiting UI here */}
    </div>
  );
};

export default WaitingPage;
