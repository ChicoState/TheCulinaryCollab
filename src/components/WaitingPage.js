import React, { useEffect, useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { auth } from '../firebase';

const WaitingPage = () => {
    const navigate = useNavigate();
    const [isEmailVerified, setIsEmailVerified] = useState(false);

    useEffect(() => {
        const interval = setInterval(async () => {
            await auth.currentUser.reload();
            if (auth.currentUser.emailVerified) {
                setIsEmailVerified(true);
            }
        }, 3000);

        return () => clearInterval(interval);
    }, []);

    useEffect(() => {
        if (isEmailVerified) {
            navigate('/workshop');
        }
    }, [isEmailVerified, navigate]);

    return (
        <div>
            <h2>Please wait while we verify your email...</h2>
        </div>
    );
};

export default WaitingPage;

