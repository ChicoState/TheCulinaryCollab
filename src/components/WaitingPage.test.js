import renderer from 'react-test-renderer';
import React, { useEffect, useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { auth } from '../firebase';
import WaitingPage from './WaitingPage';

jest.mock('react-router-dom');
jest.mock('../firebase');

const renderTree = tree => renderer.create(tree);
describe('<WaitingPage>', () => {
  it('should render component', () => {
    expect(renderTree(<WaitingPage 
    />).toJSON()).toMatchSnapshot();
  });

  
  
});