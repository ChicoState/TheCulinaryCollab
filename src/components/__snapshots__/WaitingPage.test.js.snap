// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WaitingPage> should render component 1`] = `
<div>
  <h2>
    Please wait while we verify your email...
  </h2>
</div>
`;
